File: Naiwne wyszukiwanie wzorca/ileWystapien.py
def ileWystapien(tekst, wzorzec):
    n = len(tekst)
    m = len(wzorzec)
    licznik = 0

    for i in range(n - m + 1):
        j = 0

        while j < m and tekst[i + j] == wzorzec[j]:
            j += 1

        if j == m:
            licznik += 1

    return licznik

print(ileWystapien("ababcabab", "abc"))

File: Wyszukiwanie danych Naiwne wyszukiwanie wzorca/ileWystapien.cpp
#include <iostream>
#include <string>
using namespace std;

int ileWystapien(string tekst, string wzorzec) {
    int n = tekst.length();
    int m = wzorzec.length();
    int licznik = 0;

    for(int i = 0; i <= n - m; i++) {
        int j = 0;

        while(j < m && tekst[i + j] == wzorzec[j]) {
            j++;
        }

        if(j == m) {
            licznik++;
        }
    }

    return licznik;
}